src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckClassVisitor.kt

```diff
@@ -80,17 +80,19 @@ internal class LincheckClassVisitor(
             }
         }
         fun MethodVisitor.newAdapter() = GeneratorAdapter(this, access, methodName, desc)
-        if (methodName == "<clinit>" ||
-            // Debugger implicitly evaluates toString for variables rendering
-            // We need to disable breakpoints in such a case, as the numeration will break.
-            // Breakpoints are disabled as we do not instrument toString and enter an ignored section,
-            // so there are no beforeEvents inside.
-            ideaPluginEnabled && methodName == "toString" && desc == "()Ljava/lang/String;") {
+        if (methodName == "<clinit>") {
             mv = WrapMethodInIgnoredSectionTransformer(fileName, className, methodName, mv.newAdapter())
             return mv
         }
-        if (methodName == "<init>") {
+        // Debugger implicitly evaluates toString for variables rendering
+        // We need to disable breakpoints in such a case, as the numeration will break.
+        // Breakpoints are disabled as we do not instrument toString and enter an ignored section,
+        // so there are no beforeEvents inside.
+        if (methodName == "<init>" || ideaPluginEnabled && methodName == "toString" && desc == "()Ljava/lang/String;") {
             mv = ObjectCreationTransformer(fileName, className, methodName, mv.newAdapter())
+            if (isInTraceDebuggerMode) {
```

Comment:
Don't you always need to apply `DeterministicInvokeDynamicTransformer`?

Reply:
Yes

Reply:
I see. Let's then leave a comment in the code explaining why this transformer is unnecessary.